benchexec/tools/veriabs.py

```diff
@@ -55,12 +55,10 @@ def cmdline(self, executable, options, task, rlimits):
         return [executable] + options + [task.single_input_file]
 
     def determine_result(self, run):
-        for line in run.output:
-            if "VERIABS_VERIFICATION_SUCCESSFUL" in line:
-                return result.RESULT_TRUE_PROP
-            elif "VERIABS_VERIFICATION_FAILED" in line:
-                return result.RESULT_FALSE_REACH
-            elif "NOT SUPPORTED" in line or "VERIABS_UNKNOWN" in line:
-                return result.RESULT_UNKNOWN
-
+        if run.output.any_line_contains("VERIABS_VERIFICATION_SUCCESSFUL"):
+            return result.RESULT_TRUE_PROP
+        elif run.output.any_line_contains("VERIABS_VERIFICATION_FAILED"):
+            return result.RESULT_FALSE_REACH
+        elif run.output.any_line_contains("VERIABS_UNKNOWN" or "NOT SUPPORTED"):
```

Comment:
This does not work. `"VERIABS_UNKNOWN" or "NOT SUPPORTED"` evaluates to `"VERIABS_UNKNOWN"`, so this line does not check for `NOT SUPPORTED`.

Anyway, I wonder why the change? It seems that the intention of the new code is the same as the old, so it would be only a refactoring, but the old code is not so bad that it would need to be changed. Feel free to keep it as is.

Reply:
Or do you want to change the parsing logic with regards to what happens if two or more of these result strings occur in different lines in the output? (The behavior actually does change for such logs.)

In this case, it would be good to explain such a reason in the commit message, because otherwise the knowledge about this behavior would lost in the future. Actually, it would even be better to additionally have a comment describing the desired behavior in this method if it is important.

Reply:
Yes, this is for parsing logic. Have updated the commit comment, if that works.